### DISCUSSION TITLE
[CMS] Extract more useful metadata at time of media upload

### DISCUSSION THREAD

**Discussion by rudokemper** [Terrastories/terrastories#724] [Temporarily Archived Tickets]
"For example, it would be nice to know the dimensions of videos so that the client can perform layout based on them rather than have to load the first frame of the video first before doing layout."

More discussion on this subject took place at Ruby by the Bay 2019, here; https://github.com/rubyforgood/terrastories/issues/44

**Comment by midhunkrishna** [comment]
Can take this up. Depending on the requirements, we will have to add the following dependencies: 
1. Redis, if we need to have a full fledged async adapter like Sidekiq or Resque for ActiveJob. To start off, we can add async adapter, which will execute the job in Rails's thread pool
2. ImageMagick library installed on the host
3. ffmpeg installed on the host. 

Thoughts? @rudokemper @kalimar

**Comment by kalimar** [comment]
This sounds like it can be an interesting project.   We are still a little on the fence with regards to the general architecture of the app. Is it important to have that be clear before working on this? If so we may need to wait till after hacktoberfest.

**Comment by kalimar** [comment]
@midhunkrishna how modular are those dependencies? We're still on the fence with some of the current architecture.

**Comment by midhunkrishna** [comment]
@kalimar ImageMagick and ffmpeg has to be installed via the os-es package manager and should be present on the application server. This becomes easy and can be added via Dockerfile. 

I would say we can delay redis for now, since like redis can be in itsown container, from the network or on the same application host. 

Since our app doesn't have a lot of traffic at the moment, I think we can omit redis for now.

**Comment by kalimar** [comment]
@midhunkrishna thanks for that. 

@rudokemper can you expand on what you have in mind? That will help us identify what's needed.

One last detail @midhunkrishna  this app runs entirely offline in production. Would this still work with what you are proposing?

**Comment by rudokemper** [comment]
I actually didn't originally file this one (nor really understood what it was about); it was created by somebody else during Ruby by the Bay, and I only copied the issue over to the main repo here afterwards.

**Comment by midhunkrishna** [comment]
@kalimar It should work offline. I can take this up too.

**Comment by midhunkrishna** [comment]
@kalimar Going to start on it. 👍
